doc/md/fundamentals/2-types/3-functions.md

```diff
@@ -0,0 +1,213 @@
+| `query`     | Optimized for reading data but cannot modify [state](https://internetcomputer.org/docs/motoko/fundamentals/state). |
+
+## Function comparison
+
+| Function type                | Mutates [state](https://internetcomputer.org/docs/motoko/fundamentals/state) | Calls updates | Calls queries | Asynchronous | External calls |
+|------------------------------|---------------|------------------|------------------|---------------|---------------|
+| Local                    | No            | Yes              | Yes              | No            | No            |
+| Generic                  | No            | Yes              | Yes              | No            | No            |
+| Shared update            | Yes           | Yes              | Yes              | Yes           | Yes           |
+| Shared query             | No            | No               | Yes              | Yes           | Yes           |
+| Shared composite query   | No            | No               | Yes              | Yes           | Yes           |
+
+## Local functions
+
+Local functions run within the canister's [actor](https://internetcomputer.org/docs/motoko/fundamentals/actors-async). They do not make calls to other [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters). Local functions are processed fast and synchronously.
+
+```motoko no-repl
+func add(a : Nat, b : Nat) : Nat {
+    a + b
+};
+
+let sum : Nat = add(5, 3); // Synchronous execution
+```
+
+**Example use case:** Simple computations that do not require network calls.
+
+## Generic functions
+
+Generic functions allow the use of type parameters, making them more flexible for using different data types.
+
+```motoko no-repl
+func identity<T>(x : T) : T = x;
+
+let num : Nat = identity<Nat>(42);
+let text : Text = identity<Text>("Hello");
+```
+
+## Shared functions
+
+Shared functions can be called from outside the canister (e.g., by users or other [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters)). These are processed [asynchronously](https://internetcomputer.org/docs/motoko/fundamentals/actors-async#async--await) because they interact with the network.
+
+```motoko no-repl
+  public shared func getBalance() : async Nat {
+    return 100;
+  };
+```
+
+**Example use case:** Reusable logic for different types.
+
+## Query functions
+
+[Query](https://internetcomputer.org/docs/building-apps/interact-with-canisters/query-calls) functions are designed for retrieving data. They cannot modify [state](https://internetcomputer.org/docs/motoko/fundamentals/state) and execute faster than [update](https://internetcomputer.org/docs/building-apps/interact-with-canisters/update-calls) functions because they do not go through consensus. Query functions are identified with the `query` keyword. Any function without the `query` keyword is an [update](https://internetcomputer.org/docs/building-apps/interact-with-canisters/update-calls) function.
```

Comment:
```suggestion
[Query](https://internetcomputer.org/docs/building-apps/interact-with-canisters/query-calls) functions are designed for retrieving data. They cannot permanently update [state](https://internetcomputer.org/docs/motoko/fundamentals/state) and execute faster than [update](https://internetcomputer.org/docs/building-apps/interact-with-canisters/update-calls) functions because they do not go through consensus. Query functions are identified with the `query` keyword. Any function without the `query` keyword is an [update](https://internetcomputer.org/docs/building-apps/interact-with-canisters/update-calls) function.
```